runtime/src/lib.rs

```diff
@@ -240,9 +240,30 @@ impl pallet_sudo::Config for Runtime {
 
 impl pallet_ethereum_chain_id::Config for Runtime {}
 
+/// Current approximation of the gas/s consumption considering
+/// EVM execution over compiled WASM (on 4.4Ghz CPU).
+/// Given the 500ms Weight, from which 75% only are used for transactions,
+/// the total EVM execution gas limit is: GAS_PER_SECOND * 0.500 * 0.75 => 3_000_000.
+pub const GAS_PER_SECOND: u64 = 8_000_000;
+
+/// Approximate ratio of the amount of Weight per Gas.
+/// u64 works for approximations because Weight is a very small unit compared to gas.
+pub const WEIGHT_PER_GAS: u64 = WEIGHT_PER_SECOND / GAS_PER_SECOND;
+
+pub struct MoonbeamGasWeightMapping;
+
+impl pallet_evm::GasWeightMapping for MoonbeamGasWeightMapping {
+	fn gas_to_weight(gas: usize) -> Weight {
+		Weight::try_from(gas.saturating_mul(WEIGHT_PER_GAS as usize)).unwrap_or(Weight::MAX)
+	}
+	fn weight_to_gas(weight: Weight) -> usize {
+		usize::try_from(weight.wrapping_div(WEIGHT_PER_GAS)).unwrap_or(usize::MAX)
```

Comment:
Why `wrapping_div`? Don't we want saturating here too?

Reply:
How can you do a saturating division ?

Reply:
Ahh, yeah, I was wrong here.

> Wrapping (modular) division. Computes self / rhs. Wrapped division on unsigned types is just normal division. There's no > way wrapping could ever happen. This function exists, so that all operations are accounted for in the wrapping operations.

https://doc.rust-lang.org/nightly/std/primitive.u64.html#method.wrapping_div